sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -126,26 +128,33 @@ async def __aexit__(
         await self.close()
 
     async def open(self):
-        """Opens the connection."""
-        if not self.session and self._session_owner:
-            jar = aiohttp.DummyCookieJar()
-            clientsession_kwargs = {
-                "trust_env": self._use_env_settings,
-                "cookie_jar": jar,
-                "auto_decompress": False,
-            }
-            if self._loop is not None:
-                clientsession_kwargs["loop"] = self._loop
-            self.session = aiohttp.ClientSession(**clientsession_kwargs)
-        # pyright has trouble to understand that self.session is not None, since we raised at worst in the init
-        self.session = cast(aiohttp.ClientSession, self.session)
+        if self._has_been_opened and not self.session:
+            raise ValueError(
+                "HTTP transport has already been closed. "
+                "You may check if you're calling a function outside of the `async with` of your client creation, "
+                "or if you called `await close()` on your client already."
+            )
+        if not self.session:
```

Comment:
There is still a small possibility that by the time get down to `await self.session.request()` that the transport has been closed by another thread - so I wouldn't be opposed to also adding a catch for the `AttributeError` to that call. But these changes definitely make the window for that race condition occurring much narrower, so probably not crucial.